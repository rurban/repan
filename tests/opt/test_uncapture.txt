# Test uncapture optimization

/(?|(aa)(bb)|(cc))(dd)\g1/
    uncapture, print

/(?|(aa)(bb)|(cc))(dd)\g2/
    uncapture, print

/(?|(aa)(bb)|(cc))(dd)\g3/
    uncapture, print

/\g4(?5)(?|(aa)(?|(bb)(cc)|(dd)(ee)(ff)|(gg))|(hh))(ii)/
    uncapture, print

/(?|(aa)(?|(bb)(cc)|(dd)(ee)(ff)|(gg))|(cc))(dd)(?(2)ee)(?(R1)ff)/
    uncapture, print

/(?|(?<name1>aa)(bb)(?<name2>cc)|(dd))\g{name1}/
    uncapture, print

/(?|(aa)(bb)(?<name2>cc)|(dd)(?<name1>ee)(ff)(?<name3>gg))(?&name1)/
    uncapture, print

/(?<name1>aa)(bb(?(<name2>)x)cc)(?<name3>dd)|(?<name2>ee)/u
    uncapture, print

/(?|(?<name1>aa)(?<name1>bb)(cc)(?<name1>dd)|(ee)(?<name2>ee))(?(R&name1)ff)/u
    uncapture, print

/(?|(?<name1>aa)(?<name1>bb)(cc)|(dd)(?<name2>ee)(?<name2>ff)(ee))\g{name1}(?&name2)/
    uncapture, print

