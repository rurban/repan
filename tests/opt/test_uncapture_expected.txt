# Test uncapture optimization

/(?|(aa)(bb)|(cc))(dd)\g1/
    uncapture, print
    /(?|(aa)(?:bb)|(cc))(?:dd)\g{1}/

/(?|(aa)(bb)|(cc))(dd)\g2/
    uncapture, print
    /(?|(?:aa)(bb)|(?:cc))(?:dd)\g{1}/

/(?|(aa)(bb)|(cc))(dd)\g3/
    uncapture, print
    /(?|(?:aa)(?:bb)|(?:cc))(dd)\g{1}/

/\g4(?5)(?|(aa)(?|(bb)(cc)|(dd)(ee)(ff)|(gg))|(hh))(ii)/
    uncapture, print
    /\g{1}(?2)(?|(?:aa)(?|(?:bb)(?:cc)|(?:dd)(?:ee)(ff)|(?:gg))|(?:hh))(ii)/

/(?|(aa)(?|(bb)(cc)|(dd)(ee)(ff)|(gg))|(cc))(dd)(?(2)ee)(?(R1)ff)/
    uncapture, print
    /(?|(aa)(?|(bb)(?:cc)|(dd)(?:ee)(?:ff)|(gg))|(cc))(?:dd)(?(2)ee)(?(R1)ff)/

/(?|(?<name1>aa)(bb)(?<name2>cc)|(dd))\g{name1}/
    uncapture, print
    /(?|(?<name1>aa)(?:bb)(?:cc)|(dd))\g{name1}/

/(?|(aa)(bb)(?<name2>cc)|(dd)(?<name1>ee)(ff)(?<name3>gg))(?&name1)/
    uncapture, print
    /(?|(?:aa)(bb)(?:cc)|(?:dd)(?<name1>ee)(?:ff)(?:gg))(?&name1)/

/(?<name1>aa)(bb(?(<name2>)x)cc)(?<name3>dd)|(?<name2>ee)/u
    uncapture, print
    /(?:aa)(?:bb(?(<name2>)x)cc)(?:dd)|(?<name2>ee)/

/(?|(?<name1>aa)(?<name1>bb)(cc)(?<name1>dd)|(ee)(?<name2>ee))(?(R&name1)ff)/u
    uncapture, print
    /(?|(?<name1>aa)(?<name1>bb)(?:cc)(?<name1>dd)|(ee)(?<name2>ee))(?(R&name1)ff)/

/(?|(?<name1>aa)(?<name1>bb)(cc)|(dd)(?<name2>ee)(?<name2>ff)(ee))\g{name1}(?&name2)/
    uncapture, print
    /(?|(?<name1>aa)(?<name1>bb)(cc)|(dd)(?<name2>ee)(?<name2>ff)(?:ee))\g{name1}(?&name2)/

